src/osd/PrimaryLogPG.cc

```diff
@@ -11437,6 +11437,8 @@ void PrimaryLogPG::_applied_recovered_object_replica()
 void PrimaryLogPG::recover_got(hobject_t oid, eversion_t v)
 {
   dout(10) << "got missing " << oid << " v " << v << dendl;
+  dout(10) << __func__ << " complete_to "
+           << pg_log.get_log().complete_to->version << dendl;
```

Comment:
On second thought I think it is (also) not safe to directly log the content of  the **complete_to** iter here. I post yet another fix at https://github.com/ceph/ceph/pull/23894